configure.ac

```diff
@@ -33,6 +33,17 @@ if test "x$enable_debug" = "xyes" ; then
   CXXFLAGS="$CXXFLAGS -D__SOMDebugPrint__ -Dmylockdebug -DcoherenceCheck -D__SASDebugPrint__"
 fi
 
+# Hardware Transactional Memory option
+AC_ARG_ENABLE([htm],
+  [AS_HELP_STRING([--enable-htm],
+    [Enable Hardware Transactional Memory])],
+  [],
+  [enable_htm=no])
+
+HTM=""
+AS_IF([test "x$enable_htm" != xno],
+  [AC_SUBST([HTM], [-mcpu=power8])])
```

Comment:
I don't think it's a good idea to set mcpu to power8 here.  It could cause issues on future processors that also support HTM.
What about -mhtm instead?

And that brings another question: what if an user wants to build SPHDE for a POWER7?  This change will silently add an illegal instruction to his/her code.
I recommend to automatically enable HTM code only if the compiler defaults to POWER8 or if the user set CFLAGS/CXXFLAGS, e.g.: check if the macro __HTM__ is defined by GCC at configure time.